ably/realtime_conn_spec_test.go

```diff
@@ -1078,3 +1078,64 @@ func TestRealtimeConn_RTN16(t *testing.T) {
 func sameConnection(a, b string) bool {
 	return strings.Split(a, "-")[0] == strings.Split(b, "-")[0]
 }
+
+type wrapProtoMessage struct {
+	proto.Conn
+	ok chan *proto.ProtocolMessage
+}
+
+func (w *wrapProtoMessage) Receive(deadline time.Time) (*proto.ProtocolMessage, error) {
+	msg, err := w.Conn.Receive(deadline)
+	w.ok <- msg
+	return msg, err
+}
+
+func TestRealtimeConn_RTN23(t *testing.T) {
```

Comment:
There's a very similar test called TestRealtimeConn_ReceiveTimeout. Please consider unifying both.

Reply:
I feel like they are different because we aren't really testing receive timeouts here, we are testing heartbeats.

Reply:
Per my other comments, we don't need heartbeats: we can rely on WebSocket pings and `SetReadDeadline`. Then, a receive timeout is the equivalent of a missing heartbeat.

Reply:
See https://github.com/ably/ably-go/pull/169#discussion_r456460786.